source/dub/packagesuppliers/maven.d

```diff
@@ -104,15 +104,12 @@ class MavenRegistryPackageSupplier : PackageSupplier {
 		}
 
 		auto json = Json(["name": Json(packageId), "versions": Json.emptyArray]);
-		auto xml = new DocumentParser(xmlData);
-
-		xml.onStartTag["versions"] = (ElementParser xml) {
-			 xml.onEndTag["version"] = (in Element e) {
-				json["versions"] ~= serializeToJson(["name": packageId, "version": e.text]);
-			 };
-			 xml.parse();
-		};
-		xml.parse();
+		enum versionRegEx = `<version>(.*?)<\/version>`;
```

Comment:
this doesn't parse e.g. `<version>1&#46;0&#46;6</version>` properly anymore.

I would imagine this isn't a huge deal as the SemVer version strings can all be represented easily without escaping, but this does make it no longer strictly conform to XML and servers could do anything.

I would have also expected this to change comment and whitespace behavior, however it seems the previous xml parser solution didn't care about that either, it just unescaped.

Reply:
You are right. A simple copy of the `decode` functionality from undead is about 200 lines. I wonder wheter to copy&paste this, to be on the save side?

On the other side I am not really sure decode is necessary at all here.

Reply:
> copy&paste this, to be on the save side?

Yeah, that's fine.

Reply:
Module `dub.internal.undead.xml` added. It contains the decode function and all code dependencies in the order and the format as defined in the original file.